### DISCUSSION TITLE
Consider switching from `rap2hpoutre/fast-excel` to `spatie/simple-excel`

### DISCUSSION THREAD

**Discussion by trevorgehman** [laravel/nova-issues#4521] [Q&A] [answered]
- Laravel Version: 9.19.0
- Nova Version: 4.8.0
- PHP Version: 8.1.2
- Database Driver & Version: MySQL 8.0.28

### Description:
[rap2hpoutre/fast-excel](https://github.com/rap2hpoutre/fast-excel) depends on [box/spout](https://github.com/box/spout) which is no longer maintained.

<img width="748" alt="CleanShot 2022-07-05 at 08 40 32@2x" src="https://user-images.githubusercontent.com/3209864/177341643-4f029b44-2c1c-40f4-a7a5-78c1d0f3ff7f.png">

A possible easy switch would be to use [spatie/simple-excel](https://github.com/spatie/simple-excel) which has been updated recently to use the community fork of box/spout which is [openspout/openspout](https://github.com/openspout/openspout).

**Comment by crynobone** [comment, marked as answer]
There few major reason why we not yet migrated to `spatie/simple-excel`, mainly:

1. Spatie has been aggressive with only supporting the latest PHP version, To use it with Nova 4 we need to use `^1.2.3|^2.0` and use only features that are available for all the compatible versions. 
    - Only `2.0.0` and above uses `openspout/openspout`, any projects running on Laravel 8 with PHP 7.x on Nova 4 would still depends on `box/spout`.
3. While the Reader support `LazyCollection`, the Writer doesn't and this would increase memory usage and execution time when exporting a large dataset. I would assume you would see a similar result to [Laravel Nova Export](https://docs.laravel-excel.com/nova/1.x/getting-started/).

**Comment by tonyyb** [comment]
Hello @crynobone , 

Thanks for this discussion. 

`rap2hpoutre/fast-excel` use `"openspout/openspout": "^3"` and doesn't seem very active to review PR like it : https://github.com/rap2hpoutre/fast-excel/pull/294

It is a big problem because we can't use Nova with [Livewire Powegrid](https://github.com/Power-Components/livewire-powergrid ). 

Have you thought about integrating the features of fast excel to avoid these limitations ? I think it's risky to depend of a repository that seems not very active (60 issues, 12 PR...). 

Thank you !

**Reply by crynobone** [reply]
We not moving to alternative implementation I have already explain the reason.
